mlflow/utils/databricks_utils.py

```diff
@@ -173,6 +173,13 @@ def get_job_type():
         return _get_context_tag("jobTaskType")
 
 
+def get_command_run_id():
+    try:
```

Comment:
I actually think we don't want to fall back to calling `_get_context_tag("commandRunId")`, as this will attempt to access a non-existent context tag in older runtimes. Instead, we should try-catch computing the command run ID (since it's only available in newer runtimes) and only set the `command_run_id` header if it's available/defined

Here's the backstory: the reason we have this pattern for other fields is that they're always present in older runtimes via tags on the command context (originally we'd just call e.g. `_get_context_tag("apiUrl")` to get the `apiUrl` field from the tags map in `get_webapp_url`), but we found that in particular security-enhanced runtimes, this led to errors as Python code didn't have permission to access the tags map on `CommandContext`. So in new runtimes we exposed allowlisted methods like ` apiUrl` on the `CommandContext` & in https://github.com/mlflow/mlflow/pull/3398 we added support for calling the new methods, falling back to the legacy approach of fetching values from the tags map for older runtimes.